### PR TITLE
feat: update module inputs for da solution

### PR DESCRIPTION
**Pull request by jor2**
### Description

Update module inputs in da solution to attach workload protection instance.

https://github.com/terraform-ibm-modules/terraform-ibm-scc-da/issues/23

### Release required?

- [ ] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

##### Release notes content

Update module inputs in da solution to attach workload protection instance.

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

File: solutions/instances/variables.tf [hunk 1]
```diff
@@ -202,6 +202,18 @@ variable "scc_instance_tags" {
   default     = []
 }
 
+variable "attach_wp_to_scc_instance" {
```
Comment (ocofaigh) [inline solutions/instances/variables.tf:205]: don't expose this - if `provision_scc_workload_protection` is true then the DA code should automatically set attach it to the SCC instance

File: tests/pr_test.go [hunk 2]
```diff
@@ -79,6 +79,8 @@ func TestInstancesInSchematics(t *testing.T) {
 		{Name: "scc_cos_bucket_access_tags", Value: permanentResources["accessTags"], DataType: "list(string)"},
 		{Name: "scc_wp_access_tags", Value: permanentResources["accessTags"], DataType: "list(string)"},
 		{Name: "cos_instance_access_tags", Value: permanentResources["accessTags"], DataType: "list(string)"},
+		{Name: "attach_wp_to_scc_instance", Value: true, DataType: "bool"},
```
Comment (ocofaigh) [inline tests/pr_test.go:82]: remove this since we will remove the `attach_wp_to_scc_instance` variable

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

**Comment by jor2** [comment]
https://github.com/terraform-ibm-modules/terraform-ibm-scc/pull/74

File: tests/pr_test.go [hunk 3]
```diff
@@ -79,6 +79,8 @@ func TestInstancesInSchematics(t *testing.T) {
 		{Name: "scc_cos_bucket_access_tags", Value: permanentResources["accessTags"], DataType: "list(string)"},
 		{Name: "scc_wp_access_tags", Value: permanentResources["accessTags"], DataType: "list(string)"},
 		{Name: "cos_instance_access_tags", Value: permanentResources["accessTags"], DataType: "list(string)"},
+		{Name: "provision_scc_workload_protection", Value: true, DataType: "bool"},
+		{Name: "skip_scc_wp_auth_policy", Value: false, DataType: "bool"},
```
Comment (ocofaigh) [inline tests/pr_test.go:83]: these are the default values so no need to explicitly specify here

File: solutions/instances/variables.tf [hunk 4]
```diff
@@ -202,6 +202,12 @@ variable "scc_instance_tags" {
   default     = []
 }
 
+variable "skip_scc_wp_auth_policy" {
+  type        = bool
+  default     = false
+  description = "Set to true to skip the creation of an IAM authorization policy that permits the SCC instance created by this solution read access to the workload protection instance. Only used if `attach_wp_to_scc_instance` is set to true."
```
Comment (ocofaigh) [inline solutions/instances/variables.tf:208]: remove `attach_wp_to_scc_instance` from the description - its no an exposed variable in this DA. I guess it should be replaced by `provision_scc_workload_protection`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
see comments

**Review by ocofaigh — APPROVED** [review]

**Pull request merged**